Framework/Algorithms/src/ReflectometryReductionOne2.cpp

```diff
@@ -211,6 +466,93 @@ void ReflectometryReductionOne2::exec() {
   setProperty("OutputWorkspace", IvsQ);
 }
 
+/**
+* Creates the output 1D array in wavelength from an input 2D workspace in
+* TOF. Summation is done over lambda or over lines of constant Q depending on
+* the type of reduction. For the latter, the output is projected to "virtual
+* lambda" at a reference angle twoThetaR.
+*
+* @return :: the output workspace in wavelength
+*/
+MatrixWorkspace_sptr ReflectometryReductionOne2::makeIvsLam() {
+  MatrixWorkspace_sptr result = m_runWS;
+
+  if (summingInQ()) {
+    // Currently only support a single input range of detectors
+    /// todo Generalise this to support multiple ranges
+    if (detectorGroups().size() > 1) {
+      throw std::runtime_error(
+          "Only a single range is supported for summing in Q");
+    }
+    const auto &detectors = detectorGroups()[0];
+    // Convert to lambda
+    if (m_convertUnits) {
+      g_log.debug() << "Converting input workspace to wavelength" << std::endl;
```

Comment:
good use of logging :+1: . You can avoid streaming and use `g_log.debug(your text here)`.